src/anyp/Uri.h

```diff
@@ -113,6 +113,13 @@ class Uri
     /// the provided set of expected characters.
     static SBuf Encode(const SBuf &, const CharacterSet &expected);
 
+    /// %-encode all characters except "unreserved" ones, as defined by RFC 3986
+    static SBuf Rfc3986Encode(const SBuf &);
+
+    /// %-decode characters an a buffer. Throws TextException if the input
+    /// is invalid
```

Comment:
By default, a function may throw. A parsing function that does not return std::optional value should be _expected_ to throw on parsing errors. By default, a throwing Squid code throws TextException. We do not want to start documenting all those defaults/expectations for every function, even if we have not fully embraced those defaults as developers yet.

```suggestion
    /// %-decoded version of the given buffer
```

or

```suggestion
    /// %-decodes the given buffer
```